### PR TITLE
COMMERCE-1067 - headless-batch-engine - Autogenerated

### PR DESCRIPTION
**Pull request by ivicac**
Best Regards

### CODE REVIEW

**Comment by ivicac** [comment]
ci:test:sf

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -0,0 +1,113 @@
+components:
+    schemas:
+        Task:
+            properties:
+                className:
+                    example: com.liferay.headless.delivery.dto.v1_0.BlogPosting
+                    type: string
+                endTime:
+                    example: 1569261245
```
Comment (nhpatt) [inline modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: why not a date? it's an amount in milliseconds?

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+                    format: int64
+                    minimum: 0
+                    type: integer
+                errorMessage:
+                    example: File import failed
+                    type: string
+                executeStatus:
+                    example: INITIALIZED
```
Comment (nhpatt) [inline modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: is this an enum? or it's more free-form

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+                    type: string
+                id:
+                    example: 30130
+                    format: int64
+                    minimum: 0
+                    type: integer
+                operation:
+                    example: CREATE
```
Comment (nhpatt) [inline modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: enum?

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+        get:
+            operationId: getImportTask
+            parameters:
+                - in: path
+                  explode: false
+                  name: taskId
+                  required: true
+                  schema:
+                      description: The item class API version
+                      example: v1.0
```
Comment (nhpatt) [inline modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: bad example, right? or is it a string like in the post url param

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+    title: "Headless Delivery"
+    version: v1.0
+openapi: 3.0.1
+paths:
+    /import/{taskId}/:
+        get:
+            operationId: getImportTask
+            parameters:
+                - in: path
+                  explode: false
```
Comment (nhpatt) [inline modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: builder doesn't support explode, AFAIK

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+info:
+    description: ""
+    license:
+        name: "Apache 2.0"
+        url: "http://www.apache.org/licenses/LICENSE-2.0.html"
+    title: "Headless Delivery"
+    version: v1.0
+openapi: 3.0.1
+paths:
+    /import/{taskId}/:
```
Comment (nhpatt) [inline modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: Brian is going to ask why this is not called /task/{taskId}... or import-task/importTaskId

File: modules/apps/batch-engine/headless/headless-batch-engine-api/src/main/java/com/liferay/headless/batch/engine/dto/v1_0/Body.java [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+
+/**
+ * @author Ivica Cardic
+ * @generated
+ */
+@Generated("")
+@GraphQLName("Body")
+@JsonFilter("Liferay.Vulcan")
+@XmlRootElement(name = "Body")
+public class Body {
```
Comment (nhpatt) [inline modules/apps/batch-engine/headless/headless-batch-engine-api/src/main/java/com/liferay/headless/batch/engine/dto/v1_0/Body.java]: Can it have another name? Body is a very common name and the same as the request body..

File: modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/rest-openapi.yaml [hunk 8]
```diff
@@ -44,24 +44,55 @@ info:
...
     version: v1.0
 openapi: 3.0.1
 paths:
     /import/{taskId}/:
+        delete:
+            description: Uploads a new file for deleting items in batch.
+            operationId: deleteImportTask
+            parameters:
+                - in: path
+                  name: className
```
Comment (nhpatt) [inline modules/apps/headless/headless-batch-engine/headless-batch-engine-impl/rest-openapi.yaml]: why className... you are not using it, right?

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml (hunk 1 above)
Comment (ivicac) [reply modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: It is the amount in milliseconds

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml (hunk 2 above)
Comment (ivicac) [reply modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: It is the enum. I added it to the definition.

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml (hunk 3 above)
Comment (ivicac) [reply modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: It is the enum. I added it to the definition.

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml (hunk 6 above)
Comment (ivicac) [reply modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: Changed to import-task

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml (hunk 5 above)
Comment (ivicac) [reply modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: Removed

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml (hunk 4 above)
Comment (ivicac) [reply modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: It is the right one, it has to be the same as the one defined in api.version property or @Path annotation

**Comment by ivicac** [comment]
@nhpatt Sent you updated PR: https://github.com/nhpatt/liferay-portal/pull/172

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml (hunk 1 above)
Comment (nhpatt) [reply modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: mmm we usually use dates in ISO 8601 format

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml (hunk 6 above)
Comment (nhpatt) [reply modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: 👍

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml (hunk 5 above)
Comment (nhpatt) [reply modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: 👍

File: modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml (hunk 4 above)
Comment (nhpatt) [reply modules/apps/batch-engine/headless/headless-batch-engine-impl/rest-openapi.yaml]: I meant it's modeled as an integer

**Pull request closed without merging**